### PR TITLE
Edit #2

### PR DESCRIPTION
**Pull request by romdotdog**

### CODE REVIEW

File: Understanding WebAssembly better by learning WebAssembly-Text.md [hunk 1]
```diff
@@ -135,31 +137,30 @@ that recives two `i32` as parameters and returns the largest of them.
...
 an `i32`. If it is different from `0` the instructions inside the `if..else/end` block are executed, otherwise the `else..end` is executed. If there isn't
 an `else`, simply nothing is executed.
-You might notice that the if "statement" has a result. That is done because
-when an `if` instruction is executed, the stack is marked/splited at the
+You might notice that the if "statement" has a result. That is because
+when an `if` instruction is executed, the stack is marked/split at the
 current stack top and a new isolated empty stack context is created, which we
 will call the child stack. When the `if` block is executed, the last item in
-the child stack is poped, that stack is unwinded and destroyed and the poped
+the child stack is popped, that stack is unwinded and destroyed and the poped
```
Comment (romdotdog) [inline Understanding WebAssembly better by learning WebAssembly-Text.md:156]: This part is too complex for a simple introduction.

File: Understanding WebAssembly better by learning WebAssembly-Text.md [hunk 2]
```diff
@@ -46,29 +48,30 @@ Well, the point is that WASM wasn't created to print strings and interact with
...
+All the function calls, parameter and local access are done by index. Labels
 are just compile-time annotations to make code easier to read, write and
 understand.
 
 ### Are `local.get` and `i32.add` the only instructions?
-Of course not. The WebAssembly instruction set is really big. Just the MVP
+Of course not. The WebAssembly instruction set is *huge*. Just the MVP (minimal viable product)
 has around 120 instructions. Most of them start with the type, which can only
 be `i32`, `i64`, `f32`, `f64`, followed by a dot and the name of the
 operation. There are also instructions for other purpouses, like control flow,
```
Comment (romdotdog) [inline Understanding WebAssembly better by learning WebAssembly-Text.md:77]: Mentioned control flow here, but later on you simplified a subtitle to "Decision making"

File: Understanding WebAssembly better by learning WebAssembly-Text.md [hunk 3]
```diff
@@ -209,24 +210,20 @@ Most of the things in this code have been explained. The only new things
...
  - `br_if $outer` is a conditional branch/jump instruction. If you know
-   C/C++ you know about jumping using `goto`. But using `goto` you can jump
-   every way, everywhere from every part of the code. WebAssembly is more
-   restrictive. You can branch only out. The labels are there to make 
-   the code easier to read. Without them, it would have been `br_if 1`.
-   As most of the things in WebAssembly, branching is achieved by indexes.
-   Indexing starts from the innermost to the outermost `block`. Which means
-   that the `loop` has an index `0`.
+   C/C++ you know about jumping using `goto`, but by using `goto` you can jump
+   anywhere, from anywhere in the code. WebAssembly is more
```
Comment (romdotdog) [inline Understanding WebAssembly better by learning WebAssembly-Text.md:223]: Introduce basic unconditional branching in a separate example first before hopping to a super complex conditional branching example

File: Understanding WebAssembly better by learning WebAssembly-Text.md [hunk 4]
```diff
@@ -171,8 +172,8 @@ i32.gt_s ;; doing the comparison between the two last `i32` items on the stack
 select
```
Comment (romdotdog) [inline Understanding WebAssembly better by learning WebAssembly-Text.md:172]: This code example relies too heavily on the reader's understanding of the stack

**Comment by fabricio-p** [comment]
should I put some explanation about the stack?

**Comment by fabricio-p** [comment]
@romdotdog ^

File: Understanding WebAssembly better by learning WebAssembly-Text.md (hunk 3 above)
Comment (fabricio-p) [reply Understanding WebAssembly better by learning WebAssembly-Text.md:223]: What kind of example will be better suited to simple unconditional branching

**Comment by romdotdog** [comment]
you can add another article to read as a prerequisite

File: Understanding WebAssembly better by learning WebAssembly-Text.md (hunk 3 above)
Comment (romdotdog) [reply Understanding WebAssembly better by learning WebAssembly-Text.md:223]: Some kind of example with a loop, like repeating a string

**Comment by fabricio-p** [comment]
Would it be better to talk about the linear memory before control flow?

**Comment by romdotdog** [comment]
I don't think it would be necessary, you could just mention that there is linear memory, and assume that since the reader knows what a stack is, they know what linear memory is

**Pull request merged**